diff --git a/qtbase/src/corelib/thread/qyieldcpu.h b/qtbase/src/corelib/thread/qyieldcpu.h
index 66156487d66..ad0bf205c18 100644
--- a/qtbase/src/corelib/thread/qyieldcpu.h
+++ b/qtbase/src/corelib/thread/qyieldcpu.h
@@ -20,6 +20,11 @@ extern "C"
 void _mm_pause(void);       // the compiler recognizes as intrinsic
 #endif
 
+#if defined(Q_CC_CLANG) && (__clang_major__ >= 21) && (defined(Q_PROCESSOR_ARM) || defined(Q_PROCESSOR_ARM_64))
+// Clang defines __yield() in <arm_acle.h>
+#  include <arm_acle.h>
+#endif
+
 QT_BEGIN_NAMESPACE
 
 Q_ALWAYS_INLINE
