diff --git a/qtbase/src/widgets/dialogs/qmessagebox.cpp b/qtbase/src/widgets/dialogs/qmessagebox.cpp
index f24cd905ccf..e9cab9f8cb5 100644
--- a/qtbase/src/widgets/dialogs/qmessagebox.cpp
+++ b/qtbase/src/widgets/dialogs/qmessagebox.cpp
@@ -2849,6 +2849,14 @@ bool QMessageBoxPrivate::canBeNativeDialog() const
     const QDialog * const q = static_cast<const QMessageBox*>(q_ptr);
     if (nativeDialogInUse)
         return true;
+
+#ifdef Q_OS_MACOS
+    if (QOperatingSystemVersion::current().majorVersion() >= 16) {
+        // Native message boxes on macOS 26 can't be clicked
+        return false;
+    }
+#endif
+
     if (QCoreApplication::testAttribute(Qt::AA_DontUseNativeDialogs)
         || q->testAttribute(Qt::WA_DontShowOnScreen)
         || q->testAttribute(Qt::WA_StyleSheet)
