diff --git a/qtbase/src/gui/text/qtextimagehandler.cpp b/qtbase/src/gui/text/qtextimagehandler.cpp
index 9fef9377858..e40562524ee 100644
--- a/qtbase/src/gui/text/qtextimagehandler.cpp
+++ b/qtbase/src/gui/text/qtextimagehandler.cpp
@@ -198,6 +198,7 @@ void QTextImageHandler::drawObject(QPainter *p, const QRectF &rect, QTextDocumen
         const QTextImageFormat imageFormat = format.toImageFormat();
 
     const QImage image = getImage(doc, imageFormat, p->device()->devicePixelRatio(), rect.size());
+    p->setRenderHint(QPainter::SmoothPixmapTransform);
     p->drawImage(rect, image, image.rect());
 }
 
