diff --git a/qtbase/cmake/QtPublicAppleHelpers.cmake b/qtbase/cmake/QtPublicAppleHelpers.cmake
index 1256b7f0804..c3b3d371e4a 100644
--- a/qtbase/cmake/QtPublicAppleHelpers.cmake
+++ b/qtbase/cmake/QtPublicAppleHelpers.cmake
@@ -1078,22 +1078,6 @@ function(_qt_internal_check_apple_sdk_and_xcode_versions)
         )
     endif()
 
-    if(NOT QT_NO_XCODE_MIN_VERSION_CHECK)
-        _qt_internal_get_cached_xcode_version(xcode_version)
-        if(NOT xcode_version)
-            message(FATAL_ERROR
-                    "Can't determine Xcode version. Is Xcode installed?"
-                    " Error details:\n${xcrun_error}")
-        endif()
-        if(xcode_version VERSION_LESS min_xcode_version)
-            message(${message_type}
-                "Qt requires at least version ${min_xcode_version} of Xcode, "
-                "you're building against version ${xcode_version}. Please upgrade."
-                ${extra_message}
-            )
-        endif()
-    endif()
-
     if(QT_NO_APPLE_SDK_MAX_VERSION_CHECK)
         return()
     endif()
