diff --git a/qtbase/cmake/QtAutoDetectHelpers.cmake b/qtbase/cmake/QtAutoDetectHelpers.cmake
index 6e6473da641..2fc573f01ec 100644
--- a/qtbase/cmake/QtAutoDetectHelpers.cmake
+++ b/qtbase/cmake/QtAutoDetectHelpers.cmake
@@ -329,9 +329,6 @@ function(qt_auto_detect_apple)
     _qt_internal_get_apple_sdk_version(apple_sdk_version)
     set(QT_MAC_SDK_VERSION "${apple_sdk_version}" CACHE STRING "Darwin SDK version.")
 
-    _qt_internal_get_xcode_version_raw(xcode_version_raw)
-    set(QT_MAC_XCODE_VERSION "${xcode_version_raw}" CACHE STRING "Xcode version.")
-
     if(NOT CMAKE_SYSTEM_NAME)
         # macOS
         list(LENGTH CMAKE_OSX_ARCHITECTURES arch_count)
diff --git a/qtbase/cmake/QtBaseConfigureTests.cmake b/qtbase/cmake/QtBaseConfigureTests.cmake
index ade53369b46..0b526ef8dac 100644
--- a/qtbase/cmake/QtBaseConfigureTests.cmake
+++ b/qtbase/cmake/QtBaseConfigureTests.cmake
@@ -277,7 +277,6 @@ function(qt_internal_print_cmake_darwin_info)
         message(STATUS "QT_APPLE_SDK_PATH: \"${QT_APPLE_SDK_PATH}\"")
         message(STATUS "CMAKE_OSX_DEPLOYMENT_TARGET: \"${CMAKE_OSX_DEPLOYMENT_TARGET}\"")
         message(STATUS "QT_MAC_SDK_VERSION: \"${QT_MAC_SDK_VERSION}\"")
-        message(STATUS "QT_MAC_XCODE_VERSION: \"${QT_MAC_XCODE_VERSION}\"")
 
         if(DEFINED CACHE{QT_IS_MACOS_UNIVERSAL})
             message(STATUS "QT_IS_MACOS_UNIVERSAL: \"${QT_IS_MACOS_UNIVERSAL}\"")
diff --git a/qtbase/cmake/QtPublicAppleHelpers.cmake b/qtbase/cmake/QtPublicAppleHelpers.cmake
index 3ffe53c6f34..1604150983c 100644
--- a/qtbase/cmake/QtPublicAppleHelpers.cmake
+++ b/qtbase/cmake/QtPublicAppleHelpers.cmake
@@ -928,46 +928,6 @@ function(_qt_internal_get_apple_sdk_version out_var)
     set(${out_var} "${sdk_version}" PARENT_SCOPE)
 endfunction()
 
-function(_qt_internal_get_xcode_version_raw out_var)
-    set(xcode_version "")
-    if(APPLE)
-        _qt_internal_execute_xcrun(xcode_version
-            XCRUN_ARGS xcodebuild -version
-            OUT_ERROR_VAR xcrun_error
-        )
-
-        string(REPLACE "\n" " " xcode_version "${xcode_version}")
-        string(STRIP "${xcode_version}" xcode_version)
-
-        if(NOT xcode_version)
-            message(FATAL_ERROR
-                    "Can't determine Xcode version. Is Xcode installed?"
-                    " Error details:\n${xcrun_error}")
-        endif()
-    endif()
-    set(${out_var} "${xcode_version}" PARENT_SCOPE)
-endfunction()
-
-function(_qt_internal_get_xcode_version out_var)
-    if(APPLE)
-        _qt_internal_get_xcode_version_raw(xcode_version_raw)
-
-        # The raw output is something like after the newlines are replaced with spaces:
-        # Xcode 14.3 Build version 14E222b
-        # We want only the '14.3' part. We could be more specific with the regex to match only
-        # digits separated by dots, but you never know how Apple might change the format.
-        string(REGEX REPLACE "Xcode (([^ ])+)" "\\2" xcode_version "${xcode_version_raw}")
-        if(xcode_version_raw MATCHES "Xcode ([^ ]+)")
-            set(xcode_version "${CMAKE_MATCH_1}")
-        else()
-            message(DEBUG "Failed to extract Xcode version from '${xcode_version_raw}'")
-            set(xcode_version "${xcode_version_raw}")
-        endif()
-
-        set(${out_var} "${xcode_version}" PARENT_SCOPE)
-    endif()
-endfunction()
-
 function(_qt_internal_get_cached_apple_sdk_version out_var)
     if(QT_INTERNAL_APPLE_SDK_VERSION)
         set(sdk_version "${QT_INTERNAL_APPLE_SDK_VERSION}")
@@ -979,17 +939,6 @@ function(_qt_internal_get_cached_apple_sdk_version out_var)
     set(${out_var} "${sdk_version}" PARENT_SCOPE)
 endfunction()
 
-function(_qt_internal_get_cached_xcode_version out_var)
-    if(QT_INTERNAL_XCODE_VERSION)
-        set(xcode_version "${QT_INTERNAL_XCODE_VERSION}")
-    else()
-        _qt_internal_get_xcode_version(xcode_version)
-        set(QT_INTERNAL_XCODE_VERSION "${xcode_version}" CACHE STRING "Xcode version")
-    endif()
-
-    set(${out_var} "${xcode_version}" PARENT_SCOPE)
-endfunction()
-
 # Warn or error out when the platform SDK or Xcode version are not supported.
 #
 # The messages are currently only shown when building Qt, not when building user projects
@@ -1028,7 +977,6 @@ function(_qt_internal_check_apple_sdk_and_xcode_versions)
     endif()
 
     _qt_internal_get_cached_apple_sdk_version(sdk_version)
-    _qt_internal_get_cached_xcode_version(xcode_version)
 
     if(NOT max_sdk_version MATCHES "^[0-9]+$")
         message(FATAL_ERROR
@@ -1075,14 +1023,6 @@ function(_qt_internal_check_apple_sdk_and_xcode_versions)
         )
     endif()
 
-    if(xcode_version VERSION_LESS min_xcode_version AND NOT QT_NO_XCODE_MIN_VERSION_CHECK)
-        message(${message_type}
-            "Qt requires at least version ${min_xcode_version} of Xcode, "
-            "you're building against version ${xcode_version}. Please upgrade."
-            ${extra_message}
-        )
-    endif()
-
     if(QT_NO_APPLE_SDK_MAX_VERSION_CHECK)
         return()
     endif()
