diff --git a/qtbase/src/gui/text/qtextimagehandler.cpp b/qtbase/src/gui/text/qtextimagehandler.cpp
index 70e8961467..1b246a82e6 100644
--- a/qtbase/src/gui/text/qtextimagehandler.cpp
+++ b/qtbase/src/gui/text/qtextimagehandler.cpp
@@ -133,6 +133,7 @@ void QTextImageHandler::drawObject(QPainter *p, const QRectF &rect, QTextDocumen
     Q_UNUSED(posInDocument);
         const QTextImageFormat imageFormat = format.toImageFormat();
 
+    p->setRenderHint(QPainter::SmoothPixmapTransform);
     if (QCoreApplication::instance()->thread() != QThread::currentThread()) {
         const QImage image = getAs<QImage>(doc, imageFormat, p->device()->devicePixelRatio());
         p->drawImage(rect, image, image.rect());
