diff --git a/sources/shiboken6/libshiboken/pep384impl.cpp b/sources/shiboken6/libshiboken/pep384impl.cpp
index b19b4af..2b04af8 100644
--- a/sources/shiboken6/libshiboken/pep384impl.cpp
+++ b/sources/shiboken6/libshiboken/pep384impl.cpp
@@ -738,11 +738,8 @@
 static PyTypeObject *
 getStaticMethodType(void)
 {
-    // this works for Python 3, only
-    //    "StaticMethodType = type(str.__dict__['maketrans'])\n";
     static const char prog[] =
-        "from xxsubtype import spamlist\n"
-        "result = type(spamlist.__dict__['staticmeth'])\n";
+        "result = type(str.__dict__['maketrans'])\n";
     return reinterpret_cast<PyTypeObject *>(PepRun_GetResult(prog));
 }
 
