diff --git a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py
index 9b48ab442..4905914d7 100644
--- a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py
+++ b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py
@@ -267,11 +267,6 @@ def _resolve_value(thing, valtype, line):
     if res is not None:
         type_map[thing] = res
         return res
-    warnings.warn(f"""pyside_type_init:_resolve_value
-
-        UNRECOGNIZED:   {thing!r}
-        OFFENDING LINE: {line!r}
-        """, RuntimeWarning)
     return thing
 
 
@@ -356,11 +351,7 @@ def _resolve_type(thing, line, level, var_handler, func_name=None):
         try:
             return eval(result, globals(), namespace)
         except Exception:
-            warnings.warn(f"""pyside_type_init:_resolve_type
-
-                UNRECOGNIZED:   {result!r}
-                OFFENDING LINE: {line!r}
-                """, RuntimeWarning)
+            pass
     return _resolve_value(thing, None, line)
 
 
