diff --git a/build_scripts/build_info_collector.py b/build_scripts/build_info_collector.py
index 9310c0552..f4ab3810f 100644
--- a/build_scripts/build_info_collector.py
+++ b/build_scripts/build_info_collector.py
@@ -6,9 +6,9 @@ import platform
 import sys
 from sysconfig import get_config_var
 
-from setuptools._distutils import log
-from setuptools._distutils import sysconfig as sconfig
-from setuptools._distutils.errors import DistutilsSetupError
+from distutils import log
+from distutils import sysconfig as sconfig
+from distutils.errors import DistutilsSetupError
 
 from .options import OPTION
 from .qtinfo import QtInfo
diff --git a/build_scripts/config.py b/build_scripts/config.py
index ac2cc7981..2653797db 100644
--- a/build_scripts/config.py
+++ b/build_scripts/config.py
@@ -3,7 +3,7 @@
 
 import os
 
-from setuptools._distutils import log
+from distutils import log
 
 from .versions import PYSIDE, PYSIDE_MODULE, SHIBOKEN
 
diff --git a/build_scripts/main.py b/build_scripts/main.py
index 6ce9bfe8f..7694c69cb 100644
--- a/build_scripts/main.py
+++ b/build_scripts/main.py
@@ -26,11 +26,10 @@ from setuptools.command.install import install as _install
 from setuptools.command.install_lib import install_lib as _install_lib
 from setuptools.command.install_scripts import install_scripts  # noqa: preload only
 
-# Use the distutils implementation within setuptools (but not before)
-from setuptools._distutils import log
-from setuptools._distutils import sysconfig as sconfig
-from setuptools._distutils.command.build import build as _build
-from setuptools._distutils.errors import DistutilsSetupError
+from distutils import log
+from distutils import sysconfig as sconfig
+from distutils.command.build import build as _build
+from distutils.errors import DistutilsSetupError
 
 from .build_info_collector import BuildInfoCollectorMixin
 from .config import config
diff --git a/build_scripts/options.py b/build_scripts/options.py
index 9da3beefb..11c4cc5be 100644
--- a/build_scripts/options.py
+++ b/build_scripts/options.py
@@ -1,15 +1,8 @@
 # Copyright (C) 2018 The Qt Company Ltd.
 # SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
 
-try:
-    from setuptools import Command
-    from setuptools._distutils import log
-except ModuleNotFoundError:
-    # This is motivated by our CI using an old version of setuptools
-    # so then the coin_build_instructions.py script is executed, and
-    # import from this file, it was failing.
-    from distutils import log
-    from distutils.cmd import Command
+from distutils import log
+from distutils.cmd import Command
 
 import os
 import sys
diff --git a/build_scripts/setup_runner.py b/build_scripts/setup_runner.py
index 3be0fd407..9c65657ac 100644
--- a/build_scripts/setup_runner.py
+++ b/build_scripts/setup_runner.py
@@ -7,7 +7,7 @@ import tempfile
 import textwrap
 
 from setuptools import setup  # Import setuptools before distutils
-from setuptools._distutils import log
+from distutils import log
 
 from build_scripts.config import config
 from build_scripts.main import (cmd_class_dict, get_package_version,
diff --git a/build_scripts/utils.py b/build_scripts/utils.py
index 06a9cd1f2..24fdbbed2 100644
--- a/build_scripts/utils.py
+++ b/build_scripts/utils.py
@@ -17,16 +17,8 @@ from collections import defaultdict
 from pathlib import Path
 from textwrap import dedent, indent
 
-try:
-    # Using the distutils implementation within setuptools
-    from setuptools._distutils import log
-    from setuptools._distutils.errors import DistutilsSetupError
-except ModuleNotFoundError:
-    # This is motivated by our CI using an old version of setuptools
-    # so then the coin_build_instructions.py script is executed, and
-    # import from this file, it was failing.
-    from distutils import log
-    from distutils.errors import DistutilsSetupError
+from distutils import log
+from distutils.errors import DistutilsSetupError
 
 try:
     WindowsError
diff --git a/build_scripts/wheel_override.py b/build_scripts/wheel_override.py
index e723705fc..6cb3756bd 100644
--- a/build_scripts/wheel_override.py
+++ b/build_scripts/wheel_override.py
@@ -7,7 +7,7 @@ import platform
 import sys
 from email.generator import Generator
 
-from setuptools._distutils import log as logger
+from distutils import log as logger
 
 from .options import OPTION, DistUtilsCommandMixin
 from .utils import is_64bit
diff --git a/build_scripts/wheel_utils.py b/build_scripts/wheel_utils.py
index d485aeb2f..38291785a 100644
--- a/build_scripts/wheel_utils.py
+++ b/build_scripts/wheel_utils.py
@@ -6,7 +6,7 @@ import time
 from sysconfig import get_config_var, get_platform
 
 from packaging.version import parse as parse_version
-from setuptools._distutils.errors import DistutilsSetupError
+from distutils.errors import DistutilsSetupError
 
 from .options import OPTION
 from .qtinfo import QtInfo
diff --git a/testing/wheel_tester.py b/testing/wheel_tester.py
index 8c5e5afcc..3490f83a6 100644
--- a/testing/wheel_tester.py
+++ b/testing/wheel_tester.py
@@ -32,7 +32,7 @@ this_dir = os.path.dirname(this_file)
 setup_script_dir = os.path.abspath(os.path.join(this_dir, ".."))
 sys.path.append(setup_script_dir)
 
-from setuptools._distutils import log
+from distutils import log
 
 from build_scripts.utils import (find_files_using_glob, find_glob_in_path,
                                  remove_tree, run_process, run_process_output)
diff --git a/wheel_artifacts/setup.py.base b/wheel_artifacts/setup.py.base
index 18634cbf8..9addd890c 100644
--- a/wheel_artifacts/setup.py.base
+++ b/wheel_artifacts/setup.py.base
@@ -1,6 +1,6 @@
 import setuptools
 from setuptools import setup, Extension
-from setuptools._distutils import cmd
+from distutils import cmd
 
 # This class and Extension file is intended only to force setuptools
 # to understand we are using extension modules, but because we don't
