diff --git a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py
index 199877686..0bb0ce50d 100644
--- a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py
+++ b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py
@@ -272,6 +272,7 @@ class Formatter(Writer):
 
 
 def find_imports(text):
+    import PySide6
     return [imp for imp in PySide6.__all__ if f"PySide6.{imp}." in text]
 
 
diff --git a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py
index 8c21e5d71..b36b3c96d 100644
--- a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py
+++ b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/parser.py
@@ -145,7 +145,7 @@ def _parse_line(line):
             tokens = 2 * tokens     # "self: self"
         if len(tokens) != 2:
             # This should never happen again (but who knows?)
-            raise SystemError(f'Invalid argument "{arg}" in "{line}".')
+            tokens = tokens[0], ":".join(tokens[1:])
         name, ann = tokens
         if name in keyword.kwlist:
             if LIST_KEYWORDS:
